pkg/gocachemanager/sdk.go

```diff
@@ -111,13 +116,52 @@ func (gcw *GoCacheWrapper) Get(ctx context.Context, key []byte) ([]byte, error)
 		return nil, fmt.Errorf("getting data: %w", err)
 	}
 
-	return data, nil
+	var result []byte
+
+	result, err = base64.StdEncoding.DecodeString(value)
+	if err != nil {
+		return nil, fmt.Errorf("decoding cache value: %w", err)
+	}
+
+	if gcw.gzip {
+		gzipr, err := gzip.NewReader(bytes.NewReader(result))
+		if err != nil {
+			return nil, fmt.Errorf("decompressing cache value: %w", err)
+		}
+
+		result, err = io.ReadAll(gzipr)
+		if err != nil {
+			return nil, fmt.Errorf("reading gzip bytes: %w", err)
+		}
+
+		gzipr.Close()
+	}
+
+	return result, nil
 }
 
 func (gcw *GoCacheWrapper) Set(ctx context.Context, key []byte, value []byte) error {
 	strKey := gcw.getKey(key)
 
-	return gcw.cacheManager.Set(ctx, strKey, value, store.WithExpiration(gcw.expiration))
+	var buffer bytes.Buffer
+
+	if gcw.gzip {
+		w := gzip.NewWriter(&buffer)
+
+		_, err := io.Copy(w, bytes.NewReader(value))
+		if err != nil {
```

Comment:
```suggestion
		if _, err := io.Copy(w, bytes.NewReader(value)); err != nil {
```

Always create local scoped error values if possible. No need to have it outside the if scope.